Chapter 9 questions

arnottmj opened the pull request.

Commit:
updated ask file with solution

Commit:
modidied ask solution

Commit:
added solution for old roman numeral

Commit:
updated solution for old school roman numerals

Commit:
added modern roman numeral solution

Commit:
added solution for sort

Commit:
added solution for shuffle

Commit:
removed testing code from shuffle

Commit:
added solution for dictionary sort

Commit:
added solution for english number

Commit:
added solution for 99 bottles

tansaku commented:
Hi @arnottmj thanks for submitting your work so far - much appreciated!

looking over your code, my main suggestion would be please ensure that your indenting is consistent and 2-spaced. This is a big bugbear that we have from employers - if you want a job you must ensure your code is consistently indented - it's a real red flag that will prevent you from even getting an interview in the first place. It may be that your code looks like nicely indented in your editor, but you need to ensure that you have your editor set for 2-spaced tabbing. Look at your code on github to see how the indentation is occasionally inconsistent in what you've submitted.

Please see https://github.com/bbatsov/ruby-style-guide for more details on the acceptable styles for professional coding in ruby.

Best of luck with the rest of the exercises! :-)

Commit:
updated indentation in old school roman

tansaku commented:
Hi Michael, 

Just for visibility for other team members, I'm just reposting what I sent you in slack here:

"yes the instructions in http://www.sublimetext.com/docs/indentation are just what you need (to sort your indenting), and if you are using sublime (or any other editor designed for coding) then you shouldn't have to worry about the character encoding too much"

"regarding your latest code (which I can't see - so perhaps push it to github so it will automatically be added to the pull request), I can't say precisely the issue, but if you highlight the affected area in sublime it will show with little grey dots and dashes where there are tabs and where there are spaces - sometimes you'll have tabs (or even other weird invisible characters hanging around - see my post to Kate on precourse issues: https://github.com/makersacademy/pre_course/issues/158) - but that highlighting in sublime should help to resolve the issue.  If still stuck, please please, please let's carry on the conversation in the pull request so that others can jump in given that I'm going offline for the weekend shortly ..."

Best of luck!

Commit:
updated indentation

Commit:
updated indentation

Commit:
updated indentation

arnottmj commented:
Cheers for the help. I got the problem sorted. I added the indentation settings to the appropriate preference file in sublime and then I fixed existing files by doing 'view > indentation > convert indentation to spaces." I think everything looks a bit better now.

Commit:
Merge branch 'master' of https://github.com/arnottmj/learn_to_program

Commit:
indentation finally sorted
